### ISSUE TITLE
 Type 'Balance' does not satisfy the constraint 'CompactEncodable'.

### ISSUE THREAD

**Issue by holygits** [polkadot-js/api#1882]
Seeing this error when running `tsc` appear after upgrading to `v1.1.1`

```
Type 'Balance' does not satisfy the constraint 'CompactEncodable'.
```

Complains about missing `toNumber`
```
215   public get tip (): Compact<Balance> {
                                 ~~~~~~~

  node_modules/@polkadot/types/codec/Compact.d.ts:9:5
    9     toNumber(): number;
          ~~~~~~~~~~~~~~~~~~~
    'toNumber' is declared here.
```

Seems like some typescript inference is missing

**Comment by jacogr** [comment]
Ok, that is very weird since neither of those definitions should have changed.  Can I ask you to just 

- check 1.2.1 as well (latest)
- just paste the whole error
- which tsc version

Will try and reproduce and see what is up, not sure why this has not popped in any of the polkadot-JS projects.

**Comment by jacogr** [comment]
PS: 1.2.1 does have a dependency gap (fixed in master) where it needs @types/bn.js installed as well (the dep was not listed). Just a heads up.

**Comment by holygits** [comment]
Adding `@types/bn.js` seems to be the fix. Many thanks!

**Comment by jacogr** [comment]
Ok, that issue will be solved in 1.3.1 - will get it out probably today since everybody is running into this.
